components/Common/Search/States/WithSearchBox.tsx

```diff
@@ -78,16 +89,57 @@ export const WithSearchBox: FC<SearchBoxProps> = ({ onClose }) => {
     [searchTerm, selectedFacet]
   );
 
-  const reset = () => {
-    setSearchTerm('');
-    setSearchResults(null);
-    setSelectedFacet(0);
+  useKeyboardCommands(cmd => {
+    if (searchError || !searchResults || searchResults.count <= 0) {
+      return;
+    }
+
+    switch (true) {
+      case cmd === 'down' && selectedResult === undefined:
+        setSelectedResult(0);
+        break;
+      case cmd === 'down' &&
+        selectedResult != undefined &&
+        selectedResult < searchResults.count &&
+        selectedResult < DEFAULT_ORAMA_QUERY_PARAMS.limit - 1:
+        setSelectedResult(selectedResult + 1);
+        break;
+      case cmd === 'up' && selectedResult != undefined && selectedResult != 0:
+        setSelectedResult(selectedResult - 1);
+        break;
+      case cmd === 'enter':
+        handleEnter();
+        break;
+      default:
+    }
+  });
+
+  const handleEnter = () => {
+    if (
+      selectedResult === undefined ||
+      (searchResults && searchResults?.count <= 0)
+    ) {
+      return;
+    }
+    const selectedHit = searchResults?.hits[selectedResult ?? 0];
+    if (!selectedHit) {
+      return;
+    }
+
+    router.push(searchHitToLinkPath(selectedHit));
+    handleClose();
   };
 
   const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
+
+    // Prevent searching if a result is selected
```

Comment:
In which situation could happen that a result is selected and that you hit enter on the input? Seems pretty impossible no?

Reply:
Yeah, I've just tested it to make sure of it, after selecting a result this submit does not fire, removed it